chapter4.adoc

```diff
@@ -46,57 +46,65 @@ The `ZERO` field must always be 0.
+* 0010b - `1G_allow_rw` - read and write access for the 1G range is allowed
+          for the supervisor domain
+* 0011b - `1G_allow_rwx` - read, write and execute access for the 1G range is
+          allowed for the supervisor domain
+* 0100b - `MTT_L1_DIR` - The 32M range is composed of 8192 x 4K pages
+* 0111b - `2M_PAGES` - The 32M range is composed of 16 x 2M pages
 * All other encodings are `reserved` and cause an access violation
 
 The `INFO` field depends on the `TYPE` field and is formatted as per the
 following table:
 
+[[Smmtt-rw-l2-encoding]]
 .`MTTL2` Entry Type for Smmtt
 [width="100%",cols="25%,75%",options="header",]
 |===
 |*`MTTL2` Entry `TYPE`* |*Description, `INFO` and `TYPE` field encoding*
 |`1G_disallow` a|
-_The 1G range of address is not allowed for the domain._
+_read, write or execute is not allowed to this 1G address range for the domain._
+The `INFO` field must be 0.
+When configuring 1G ranges, RDSM ensures that 16 `MTTL2` entries, each
+corresponding to 64M of address space, have identical `TYPE` field values.
+
+|`1G_allow_rx` a|
+_read and execute (but no write) are allowed to this 1G address range for the
+domain._
 The `INFO` field must be 0.
 When configuring 1G ranges, RDSM ensures that 32 `MTTL2` entries, each
 corresponding to 32M of address space, have identical `TYPE` field values.
 
-|`1G_allow_r` a|
-_The 1G range of address is allowed (read only) for the domain._
+|`1G_allow_rw` a|
+_read and write (but no execute) are allowed to this 1G address range for the
+domain._
 The `INFO` field must be 0.
 When configuring 1G ranges, RDSM ensures that 32 `MTTL2` entries, each
 corresponding to 32M of address space, have identical `TYPE` field values.
-The INFO field must be 0.
 
-|`1G_allow_rw` a|
-_The 1G range of address is allowed (read/write) for the domain._
+|`1G_allow_rwx` a|
+_read, write and execute are allowed to this 1G address range for the domain._
 The `INFO` field must be 0.
 When configuring 1G ranges, RDSM ensures that 32 `MTTL2` entries, each
 corresponding to 32M of address space, have identical `TYPE` field values.
 
 |`MTT_L1_DIR` a|
 The `INFO` field provides the PPN of the `MTTL1` page.
-Entries of the `MTTL1` page hold a 4-bit `PERM` (permissions) field to indicate
-the access for the supervisor domain (described in the `MTTL1` entry
-<<mtt-l1-rw>>).
+Entries of the `MTTL1` page hold a 4-bit `PERM` field to indicate the access
+for the supervisor domain (described in the `MTTL1` entry <<mtt-l1-rw>>).
 
 |`2M_PAGES` a|
 _The 32M range of address space is partitioned into 2M pages where each
 page has read/write access allowed/not._
 The `INFO` field 31:0 holds 2 `PERM` bits per 2M address range to indicate
-`access_disallowed` (00b), `read_allowed` (01b), `read_write_allowed` (11b);
-(10b is `reserved`). Bits 32:43 are reserved and must be zero.
+`rwx_disallowed` (00b), `rx_allowed` (01b), `rw_allowed` (10b), `rwx_allowed`
+(11b). Bits 32:43 are reserved and must be zero.
```

Comment:
Indicate that other INFO field bits must be 0? When access is not allowed  it causes access violation.

Reply:
rwx_disallowed -> access disallowed?

Reply:
It may not be obvious what "rx_allowed" means. Maybe use same language as used for L1 access permission encodings description?

Reply:
fixed